### ISSUE TITLE
`siblings query` slow

### ISSUE THREAD

**Issue by mih** [datalad/datalad#7090]
https://github.com/datalad/datalad-gooey/issues/365 talks about a runtime of "minutes"

**Comment by bpoldrack** [comment]
> one could argue that this kind of expensive query should not happen by default.

Yes, agree. That is unbearable, especially when a siblings query is used in order to feed into something else (which then would appear hanging for minutes). Just like `git remote`, `datalad siblings` should just list the locally known stuff by default, not "analyze" it.

**Comment by yarikoptic** [comment]
> Just like `git remote`, `datalad siblings` should just list the locally known stuff by default, not "analyze" it.

I would think so too.

**Comment by yarikoptic** [comment]
BTW, to some degree it then gets to the same aspect you @bpoldrack mentioned in the https://github.com/datalad/datalad/pull/7071#issuecomment-1272516296 :

> I would like to not conflate the "output stream of a command" with its rendered results. Those are two different concepts. While rendered results can be part of the output stream of a command, that's not generally true and an output stream of a command can contain things that are not rendered results (as is the case with --o-s pass-through).

In that for speedy output of just names of the siblings - there is not much to do. But if your output wants to provide extra information, then output might require the mode of operation which would provide that information. Similarly here -- if output cares only about the names, no need to query for annex info.
But that means also that default mode of operation of all commands should strive to be the leanest with options making it "heavier".   That is not what a philosophy of git and git-annex though, consider `annex` having `--fast` option and `git remote` having `-n`. So, unfortunately, nothing pure out there but most things are made "pragmatic" in their default mode of operation to cover most of the use cases.

Thus getting back to this issue, and given that typically we have only few siblings, if gooey is interested just in speedy "names only" I guess it could ask to disable fetching annex info.

**Comment by mih** [comment]
Would be good to know, if anybody is aware of the use case that makes "heavy" the default.
